R/helpers_jags.R

```diff
@@ -122,25 +122,30 @@ h_jags_get_data <- function(model, data, from_prior) {
   assert_class(data, "GeneralData")
   assert_flag(from_prior)
 
+  # 1) Extract variables from `data` as required by `modelspecs`.
   modelspecs <- do.call(
     model@modelspecs,
     h_slots(data, formalArgs(model@modelspecs))
   )
   assert_list(modelspecs)
 
-  data_model <- if (from_prior) {
-    # Remove elements named "ref_dose" to avoid JAGS error of unused variables.
-    modelspecs <- modelspecs[setdiff(names(modelspecs), "ref_dose")]
-    NULL
+  # 2) Extract variables from `data` as required by `datanames`.
+
+  datanames <- if (from_prior) {
+    # This is a workarround to avoid JAGS error of unused variables.
```

Comment:
Hi @danielinteractive  - Please note this workaround. It was here before for `ref_dose` only, I added also `use_log_dose` coming from `DualEndpoint` models. I think this is also something that should be addressed.